### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by melekes**
Fixes:
- #3307
- #536 
- #2904

### CODE REVIEW

File: docs/DOCS_README.md [hunk 1]
```diff
@@ -66,11 +66,12 @@ To build and serve the documentation locally, run:
...
 ```
 # from this directory
-npm install
 npm install -g vuepress
 ```
 
-then change the following line in the `config.js`:
+NOTE: the command may require `sudo`.
+
+then change the following line in the `.vuepress/config.js`:
```
Comment (zramsay) [inline docs/DOCS_README.md]: lets make this change in the same file in the SDK repo

File: rpc/core/mempool.go [hunk 2]
```diff
@@ -16,7 +16,8 @@ import (
 //-----------------------------------------------------------------------------
 // NOTE: tx should be signed, but this is only checked at the app level (not by Tendermint!)
 
-// Returns right away, with no response
+// Returns right away, with no response. Does not wait for CheckTx nor
```
Comment (zramsay) [inline rpc/core/mempool.go]: note to self; update slate once this is merged

File: docs/tendermint-core/configuration.md [hunk 3]
```diff
@@ -268,3 +268,22 @@ max_open_connections = 3
...
 namespace = "tendermint"
 ```
+
+## Empty blocks VS no empty blocks
+
+**create_empty_blocks = true**
+
+If `create_empty_blocks` is set to `true` in your config, blocks will be
+created ~ every second (with default consensus parameters which are explained
+[here](https://forum.cosmos.network/t/consensus-timeouts-explained/1421)). You
```
Comment (ebuchman) [inline docs/tendermint-core/configuration.md]: Lol we should bring that explanation home too

File: docs/tendermint-core/configuration.md [hunk 4]
```diff
@@ -268,3 +268,22 @@ max_open_connections = 3
...
+can regulate the delay between blocks by changing the `timeout_commit`. E.g.
+`timeout_commit = "10s"` should result in ~ 10 second blocks.
+
+**create_empty_blocks = false**
+
+In this setting, blocks are created when txs received. Note after the block N,
+Tendermint creates something we call a "proof block" (if the app hash changes)
+N+1 [#2487 (comment)](https://github.com/tendermint/tendermint/issues/2487#issuecomment-424899799).
```
Comment (ebuchman) [inline docs/tendermint-core/configuration.md]: Will this link render probably? Maybe we can just explain a bit better here?

File: docs/tendermint-core/configuration.md [hunk 5]
```diff
@@ -268,3 +268,22 @@ max_open_connections = 3
...
+Plus, if you set `create_empty_blocks_interval` to something other than the
+default (`0`), Tendermint will be creating empty blocks even in the absence of
+transactions every `create_empty_blocks_interval`.
```
Comment (ebuchman) [inline docs/tendermint-core/configuration.md]:
```suggestion
transactions every `create_empty_blocks_interval`. For instance, with `create_empty_blocks = false` and `create_empty_blocks_interval = "30s"`, Tendermint will only create blocks if there are txs, or after waiting 30s without receiving any txs.
```

File: rpc/core/mempool.go [hunk 6]
```diff
@@ -98,6 +143,50 @@ func BroadcastTxAsync(tx types.Tx) (*ctypes.ResultBroadcastTx, error) {
 // | Parameter | Type | Default | Required | Description     |
 // |-----------+------+---------+----------+-----------------|
 // | tx        | Tx   | nil     | true     | The transaction |
+//
+// To send a UTF8 string byte array, you simply quote the value of the tx parameter on the GET request:
```
Comment (ebuchman) [inline rpc/core/mempool.go]: Can we deduplicate somehow? Will slate let us link to a previous methods description or to some comment documentation?

File: rpc/core/mempool.go [hunk 7]
```diff
@@ -53,6 +54,50 @@ import (
...
+//
+//     GET http://myserver:46657/broadcast_tx_async?tx="hello"
+//
+// This sends a 5 byte transaction "h e l l o" [68 65 6c 6c 6f]
+//
+//     GET http://myserver:46657/broadcast_tx_async?tx="5"
+//
+// This sends a 4 byte transaction "5" (UTF8) [e2 82 ac 35]
```
Comment (ebuchman) [inline rpc/core/mempool.go]: Huh? `"5"` is 4 bytes?!

File: rpc/core/mempool.go [hunk 8]
```diff
@@ -53,6 +54,50 @@ import (
...
+//
+// ```
+// {
+//   "method": "broadcast_tx_async",
+//   "jsonrpc": "2.0",
+//   "params": {"tx": "0x01020304"},
+//   "id": ""
+// }
+// ```
+//
```
Comment (ebuchman) [inline rpc/core/mempool.go]: Lol this is really something :D

File: rpc/core/mempool.go (hunk 6 above)
Comment (melekes) [reply rpc/core/mempool.go]: I think it's totally fine to duplicate (until we have better RPC docs / switch to swagger https://github.com/tendermint/tendermint/issues/2419) and not force a person to go somewhere else.

Comment (melekes) [reply rpc/core/mempool.go]: Actually, looks like we already had this info in "Using Tendermint". Replaced with the link

File: rpc/core/mempool.go (hunk 7 above)
Comment (melekes) [reply rpc/core/mempool.go]: mistake

**Comment by melekes** [comment]
Thanks for review guys! Fixed the comments. PTAL

**Pull request merged**